### PR TITLE
fix: Performance fix to only measure statistics which are not hidden

### LINKED ISSUE #177: Only measure and issue for the columns that are reported
Currently in the `get_per_issue_metrics()` function, it measure things like time to first response, time to answer, and time to close, even if those columns are marked as hidden via `HIDE_TIME_TO_FIRST_RESPONSE = true`

### PR DESCRIPTION
**Pull request by zkoppert**
fixes #177

# Pull Request

Please review cautiously as my confidence around these changes is low as I was in a rush. Happy to pair on a review.

## Proposed Changes

This pull request primarily focuses on enhancing the functionality of the `issue_metrics.py` script by introducing a new `EnvVars` class and using it to control the execution of certain metrics calculations. The changes allow for more flexibility in terms of what metrics to calculate and when.

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing

### Reviewer

- [x] Label as either `bug`, `documentation`, `enhancement`, `infrastructure`, or `breaking`

### CODE REVIEW

**Comment by zkoppert** [comment]
@jmeridth Any ideas on how to handle a default argument for the EnvVars type?
>issue_metrics.py:130: error: Incompatible default for argument "env_vars" (default has type "None", argument has type "EnvVars")  [assignment]

**Comment by jmeridth** [comment]
> @jmeridth Any ideas on how to handle a default argument for the EnvVars type?
> 
> > issue_metrics.py:130: error: Incompatible default for argument "env_vars" (default has type "None", argument has type "EnvVars")  [assignment]

```diff
diff --git a/issue_metrics.py b/issue_metrics.py
index 628f7f3..85c3b39 100644
--- a/issue_metrics.py
+++ b/issue_metrics.py
@@ -127,7 +127,7 @@ def get_per_issue_metrics(
     ignore_users: Union[List[str], None] = None,
     max_comments_to_eval: int = 20,
     heavily_involved: int = 3,
-    env_vars: EnvVars = None,
+    env_vars: EnvVars = get_env_vars(),
 ) -> tuple[List, int, int]:
     """
     Calculate the metrics for each issue/pr/discussion in a list provided.
```

You could do some funky stuff with `Union[EnvVars,None]` as the return type from `get_env_vars()` but then you have to test for None everywhere it is used.  Above is "easier". wdyt?

**Comment by zkoppert** [comment]
@jmeridth ready for next round of review when you have time

File: .github/scripts/env_vars_check.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    # If any instances are found, print the file name and line number
+    if [ -n "$result" ]; then
+        echo "Found in $file:"
+        echo "$result"
+        echo -e "${RED}ERROR: get_env_vars() should always set test=True in test*.py files.${NC}"
+        exit 1
+    fi
+done
+echo -e "   ${GREEN}PASS:${NC} All test*.py files call get_env_vars() with test=True."
```
Comment (jmeridth) [inline .github/scripts/env_vars_check.sh:22]: small nit: newline at end of file pretty please

**Review by jmeridth — APPROVED** [review]
LGTM just one nit.

**Review by jmeridth — APPROVED** [review]

**Pull request merged**